### PR TITLE
Testing app

### PR DESCRIPTION
**Pull request by abeidahmed**
- Add tests
  - Run `npm run test` to run all the tests
  - Run `npm run test:watch` to watch the test
- Add GitHub action for running the tests

### CODE REVIEW

File: src/tests/logic/calculate.spec.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import calculate from '../../logic/calculate';
```
Comment (Torres-ssf) [inline src/tests/logic/calculate.spec.js:1]: - [x] We need to add more tests here to test all calculator operators: +, -, *, /, %. For example:
expect that `5 * 20` will result in 100

File: src/tests/logic/calculate.spec.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        operation: 'n',
+      },
+      '-',
+    );
+
+    expect(newData).toEqual({ next: null, operation: null });
+  });
+
+  it('returns valid object when button is %', () => {
+    const newData = calculate(data, '%');
```
Comment (Torres-ssf) [inline src/tests/logic/calculate.spec.js]: - [x] We need to test if the modifier operator is working as expected, like:
`Before: 5`
`After: 0.05`

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes Required :muscle: 
Hello @abeidahmed :hand: 
First of all, awesome job in creating tests for your calculator app. 
You did a nice job also creating Snapshots. Congratulations :clap: :tada: 
However, we need to test all the calculator functionalities.

- [x] Check my comments below.

Happy coding :computer: 
**Sérgio Torres**

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @abeidahmed :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Comment by abeidahmed** [comment]
> ## Status: Approved 
> ### Hello @abeidahmed 
> ### Thank you for making all of the requested changes. Your project is approved   
> ### You may merge, good luck with your next projects 
> Murilo Roque Paiva da Silva
> 
> * Github: [@muriloroque](https://github.com/MuriloRoque)
> * Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)
> 
> [Twitter Follow](https://twitter.com/MuriloRoquePai1)

Thanks @MuriloRoque

**Comment by abeidahmed** [comment]
> ### Changes Required 
> Hello @abeidahmed 
> First of all, awesome job in creating tests for your calculator app.
> You did a nice job also creating Snapshots. Congratulations  
> However, we need to test all the calculator functionalities.
> 
> * [x]  Check my comments below.
> 
> Happy coding 
> **Sérgio Torres**

Thanks, @Torres-ssf for the feedback!

**Pull request merged**